public/js/components/Autocomplete.js

```diff
@@ -51,7 +73,11 @@ const Autocomplete = React.createClass({
       dom.input(
         {
           ref: "searchInput",
-          onChange: (e) => this.setState({ inputValue: e.target.value })
+          onChange: (e) => this.setState({
+            inputValue: e.target.value,
+            selectedIndex: -1
+          }),
```

Comment:
I agree, but then it cannot be index-based. There needs to be a reference to the actual source or an identifier of it.